src/main/java/com/aws/greengrass/logmanager/util/FileHashGenerator.java

```diff
@@ -0,0 +1,63 @@
+package com.aws.greengrass.logmanager.util;
+
+import com.aws.greengrass.logging.api.Logger;
+import com.aws.greengrass.logging.impl.LogManager;
+import com.aws.greengrass.logmanager.LogManagerService;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.security.NoSuchAlgorithmException;
+import java.util.Optional;
+
+import static com.aws.greengrass.util.Digest.calculate;
+
+public final class FileHashGenerator {
+    private static final Logger logger = LogManager.getLogger(LogManagerService.class);
+    // private constructor to make checkstyle happy
+    private FileHashGenerator() {
+        
+    }
+
+    /**
+     * calculate the hash for the log file, which takes and digests first some lines (default is 1) in the file.
+     * If the content of file is insufficient, the hash is null.
+     * @param logFile
+     * @return the nullable calculated hash for active file
+     */
+    public static Optional<String> calculateHashForLogFile(File logFile, int DEFAULT_LINES_FOR_DIGEST_NUM) {
+        Optional fileHash = Optional.ofNullable(null);
+        // empty file, then do not read
+        if (!logFile.isFile())
```

Comment:
This does not check if the file is empty. It checks if it is a directory or a file.